### ISSUE TITLE
Allow Disabling of Editor Command Palette

### ISSUE THREAD

**Issue by MadtownLems** [WordPress/gutenberg#62337]
## What problem does this address?

6.6-beta-1 looks to include the Command Palette in the Post Editor. This addition looks to be very confusing for our users, and provides very little - if any - benefit. We'd really like to not have to educate our large number of content providers on it, especially if the education ends up being: "Just ignore it"

## What is your proposed solution?

Add a filter, ideally through block_editor_settings_all (https://developer.wordpress.org/reference/hooks/block_editor_settings_all/) that lets users easily disable the Editor Command Palette.

**Comment by annezazu** [comment]
Just a quick note that the command palette has been available in the post editor for quite some time ([since 6.3](https://make.wordpress.org/core/2023/07/17/introducing-the-wordpress-command-palette-api/)). It's not new to 6.6.

**Comment by MadtownLems** [comment]
> Just a quick note that the command palette has been available in the post editor for quite some time

Correct - but 6.6 makes it prominently displayed at the top of the post editor. This will definitely be the first time that any of our content providers are aware of it, and shown to hit Ctrl+K to open it

![2024-06-11 12_07_03-Window](https://github.com/WordPress/gutenberg/assets/945157/3f7aa7fb-a0da-42fe-a74b-bc22abf58f06)

**Comment by annezazu** [comment]
Yes! That's part of the work done here with 6.6: https://github.com/WordPress/gutenberg/issues/52632 It's important to note though what's new for each release when deciding what can be fixed or changed in this beta period hence the note from yours truly :)

**Comment by ndiego** [comment]
> Add a filter, ideally through block_editor_settings_all (https://developer.wordpress.org/reference/hooks/block_editor_settings_all/) that lets users easily disable the Editor Command Palette.

I agree. There should be Editor setting that controls whether the Command Palette is available or not. This setting should default to `true`, but allow developers to disable it in specific Editor contexts.

**Comment by MadtownLems** [comment]
To complicate this issue, I learned that the only way to go "Back" from "Edit Original" pattern exists in the Command Palette.  If disabling the Command Palette is supported, there should be an alternative method to go 'back' to the page/post you were working on in that context.

**Comment by petertwise** [comment]
How did the visible top admin bar link to this feature make it into 7.0 without a more sophisticated way to disable it or contextually filter it?

**Comment by MadtownLems** [comment]
> How did the visible top admin bar link to this feature make it into 7.0 without a more sophisticated way to disable it or contextually filter it?

This link in the Admin Bar is quite easy to remove using core API's: https://developer.wordpress.org/reference/classes/wp_admin_bar/remove_node/

**Comment by petertwise** [comment]
Sure, - or you could also hide it with `#wp-admin-bar-command-palette {display:none;}` ... but that doesn't remove the functionality. The best we've been able to do for that is something like this:

```php
// Disable WP 7.0+ Command Palette
add_action( 'init', 'squarecandy_disable_command_palette' );
function squarecandy_disable_command_palette() {
	remove_action( 'admin_enqueue_scripts', 'wp_enqueue_command_palette_assets' );
	remove_action( 'admin_bar_menu', 'wp_admin_bar_command_palette_menu', 55 );
}
```

Plus I think the underlying issue prompting people to ask to disable this for the editor level specifically is that it's not well integrated into permissions system. If a plugin's settings screen URL returns "Sorry you're not allowed to access this page" it shouldn't show up in the list of possible actions in the command palette. If a better sub-set of the possible results based on each user role were displayed, there would be less reason to need to hide it or disable it for user roles lower than full Admin.

**Comment by Brugman** [comment]
Thanks @petertwise . The admin_bar_menu hook seems to be unnecessary at the moment. The site editor title bar still shows ctrl+k so that needs a little CSS. For whoever cares: https://gist.github.com/Brugman/789109bbaeac30bd60828a7fcfbb981a

Looking forward to a command palette that only shows things that the user role actually has the capabilities for.
